hostapd: update to version 2025-02-09

rany2 opened the pull request:
Manually refreshed:

- 010-mesh-Allow-DFS-channels-to-be-selected-if-dfs-is-ena.patch
- 110-mbedtls-TLS-crypto-option-initial-port.patch
- 140-tests-Makefile-make-run-tests-with-CONFIG_TLS.patch
- 301-mesh-noscan.patch
- 601-ucode_support.patch
- 780-Implement-APuP-Access-Point-Micro-Peering.patch

Dropped upstreamed:

- 330-nl80211_fix_set_freq.patch
- 804-hostapd-Fix-clearing-up-settings-for-color-switch.patch

Automatically rebased all other patches.

Neustradamus commented:
@rany2: Nice PR, thanks!

rany2 commented:
@dhewg thanks, I've added you to the commit description :)

e9hack commented:
Something is broken in hostapd. hostapd crashes if a client connect via wpa enterprise to a 5G WiFi:
```
Wed Jan  1 09:51:44 2025 daemon.info hostapd: 2G-ap4: STA aa:aa:aa:aa:aa:aa IEEE 802.11: authenticated
Wed Jan  1 09:51:44 2025 daemon.info hostapd: 2G-ap4: STA aa:aa:aa:aa:aa:aa IEEE 802.11: associated (aid 19)
Wed Jan  1 09:51:44 2025 daemon.notice hostapd: Assigned VLAN ID 30 from wpa_psk_file to aa:aa:aa:aa:aa:aa
Wed Jan  1 09:51:44 2025 daemon.notice hostapd: 2G-ap4: AP-STA-CONNECTED aa:aa:aa:aa:aa:aa auth_alg=open
Wed Jan  1 09:51:44 2025 daemon.info hostapd: 2G-ap4: STA aa:aa:aa:aa:aa:aa RADIUS: starting accounting session F05DC2209110C2A3
Wed Jan  1 09:51:44 2025 daemon.info hostapd: 2G-ap4: STA aa:aa:aa:aa:aa:aa WPA: pairwise key handshake completed (RSN)
Wed Jan  1 09:51:44 2025 daemon.notice hostapd: 2G-ap4: EAPOL-4WAY-HS-COMPLETED aa:aa:aa:aa:aa:aa
Wed Jan  1 09:52:56 2025 daemon.info hostapd: 5G-ap1: STA xx:xx:xx:xx:xx:xx IEEE 802.11: authenticated
Wed Jan  1 09:52:56 2025 daemon.info hostapd: 5G-ap1: STA xx:xx:xx:xx:xx:xx IEEE 802.11: associated (aid 2)
Wed Jan  1 09:52:56 2025 daemon.notice hostapd: 5G-ap1: CTRL-EVENT-EAP-STARTED xx:xx:xx:xx:xx:xx
Wed Jan  1 09:52:57 2025 daemon.notice hostapd: 5G-ap1: CTRL-EVENT-EAP-PROPOSED-METHOD vendor=0 method=1
Wed Jan  1 09:53:00 2025 daemon.notice hostapd: 5G-ap1: CTRL-EVENT-EAP-RETRANSMIT xx:xx:xx:xx:xx:xx
Wed Jan  1 09:53:06 2025 daemon.notice hostapd: 5G-ap1: CTRL-EVENT-EAP-RETRANSMIT xx:xx:xx:xx:xx:xx
Wed Jan  1 09:53:18 2025 daemon.notice hostapd: 5G-ap1: CTRL-EVENT-EAP-RETRANSMIT xx:xx:xx:xx:xx:xx
Wed Jan  1 09:53:38 2025 daemon.notice hostapd: 5G-ap1: CTRL-EVENT-EAP-RETRANSMIT xx:xx:xx:xx:xx:xx
Wed Jan  1 09:53:58 2025 daemon.notice hostapd: 5G-ap1: CTRL-EVENT-EAP-RETRANSMIT xx:xx:xx:xx:xx:xx
Wed Jan  1 09:54:17 2025 daemon.info hostapd: 5G-ap1: STA xx:xx:xx:xx:xx:xx IEEE 802.11: authenticated
Wed Jan  1 09:54:17 2025 daemon.info hostapd: 5G-ap1: STA xx:xx:xx:xx:xx:xx IEEE 802.11: associated (aid 2)
Wed Jan  1 09:54:18 2025 daemon.notice hostapd: 5G-ap1: CTRL-EVENT-EAP-RETRANSMIT xx:xx:xx:xx:xx:xx
Wed Jan  1 09:54:18 2025 daemon.notice hostapd: 5G-ap1: CTRL-EVENT-EAP-TIMEOUT-FAILURE xx:xx:xx:xx:xx:xx
Wed Jan  1 09:54:20 2025 kern.info kernel: [30197.810443] do_page_fault(): sending SIGSEGV to hostapd for invalid read access from 087f000c
Wed Jan  1 09:54:20 2025 kern.info kernel: [30197.819092] epc = 5559c9ad in hostapd-50af5c1c[55560000+cb000]
Wed Jan  1 09:54:20 2025 kern.info kernel: [30197.825077] ra  = 555971f1 in hostapd-50af5c1c[55560000+cb000]
Wed Jan  1 09:54:20 2025 daemon.notice netifd: Network device '5G-ap2' link is down
Wed Jan  1 09:54:20 2025 kern.info kernel: [30197.843331] br-tor: port 2(5G-ap2) entered disabled state
Wed Jan  1 09:54:20 2025 kern.info kernel: [30197.926864] br-tor: port 2(5G-ap2) entered disabled state
```
This is currently hostapd from 31.12., but it occurs with older versions too. I think it did start with hostapd versions from mid of December (18.12. ??) and similar updated patches.

nbd168 commented:
@e9hack can you please grab the corefile from the device and use ./scripts/remote-gdb in the openwrt tree to obtain a backtrace?
Before you use remote-gdb, I'd recommend doing `make package/hostapd/{clean,compile} CONFIG_DEBUG=y` (no need to update hostapd on the device, it's only for local debug data).

e9hack commented:
I don't got a core file from hostapd. From other programs I got core files on a crash (dnsmasq, ujail, nft, awk, netifd, ...). Currently I start hostapd via two symlinks to be able to recover in an easy way:
```
root@WLAN-DSL9:/usr/sbin# ls -l h*
lrwxrwxrwx    1 root     root            27 Dec 31 19:22 hostapd -> /mnt/1/test/hostapd/hostapd
-rwxr-xr-x    1 root     root        865495 Dec 31 17:44 hostapd.old

root@WLAN-DSL9:/mnt/1/test/hostapd# ls -l h*
lrwxr-xr-x    1 root     root            16 Jan  1 11:27 hostapd -> hostapd-50af5c1c
-rwxr-xr-x    1 root     root        865495 Dec 31 19:18 hostapd-50af5c1c
```
I'm not sure if this is the reason why there is no core file.

rany2 commented:
@e9hack can you try now? if it doesn't help, what hostapd variant are you using right now?

rany2 commented:
I can't trigger a SEGFAULT when connecting to a WPA Enterprise AP.

@e9hack What EAP method are you using and which TLS backend are you using?

e9hack commented:
EAP method is PEAP/MSCHAP2 (Windows 10 client) and TLS backend is openssl.
I got again a crash without a core dump:
```
Wed Jan  1 13:24:33 2025 kern.info kernel: [  434.352904] do_page_fault(): sending SIGSEGV to hostapd for invalid read access from 0000000c
Wed Jan  1 13:24:33 2025 kern.info kernel: [  434.361517] epc = 555cc9ad in hostapd[55590000+cb000]
Wed Jan  1 13:24:33 2025 kern.info kernel: [  434.366613] ra  = 555c71f1 in hostapd[55590000+cb000]
```
Do I need additional settings to get a core dump?

nbd168 commented:
@e9hack  I guess an alternative would be to put gdbserver on your device, attach it to the hostapd process, use remote-gdb to connect to it and then trigger a crash.

e9hack commented:
I've no success with gdbserver and remote-gdb. No page fault did occur when remote-gdb is connected. I'm not sure if remote-gdb does properly work. If hostapd was restarted and I did kill and restart gdbserver, remote-gdb doesn't dectecd this.

Indepenendly of this, it looks like that hostapd commit 

01677c47fb13976e078b5a2157aa6bebf19f1731
AP: Support disconnect with MLD

from 19.12.24 does trigger my issue. It is possible that one of the patches is the real trigger. Starting with this commit if I shutdown or reboot a connected Windows 10 client, hostapd is restarted, sometimes with a page fault sometimes without. Only disconnecting the windows 10 client does not trigger the issue all the time. My windows 10 client is connect to the 5G WiFi. I didn't test it with 2G WiFi yet. When such a restart of hostapd occurs, all ESP8286 IoT clients will frequently connect and disconnect from the 2G WiFi. The router must be restart to get them connected permanently again.

rany2 commented:
@e9hack Can you make a post on the mailing list regarding this? I think if nobody comes up with a proper fix I could just revert the commit causing your issue.

rany2 commented:
@e9hack Just a shot in the dark but I noticed that `050-Fix-OpenWrt-13156.patch` might be interacting negatively with the upstream commit you specified (01677c47fb13976e078b5a2157aa6bebf19f1731 AP: Support disconnect with MLD). Can you try dropping `050-Fix-OpenWrt-13156.patch` to see if it resolves the issue?

danpawlik commented:
On BPi-R4 with mt7996, there are some strange messages:
```
Tue Jan  7 09:34:14 2025 daemon.notice hostapd: nl80211: Failed to remove interface phy0.0-ap0 from bridge br-lan: Invalid argument
Tue Jan  7 09:34:15 2025 daemon.err hostapd: Failed to request a scan of neighboring BSSes ret=-16 (Resource busy) - try to scan again (attempt 2)
Tue Jan  7 09:34:15 2025 kern.info kernel: [   39.675847] br-lan: port 7(phy0.0-ap0) entered blocking state
Tue Jan  7 09:34:15 2025 kern.info kernel: [   39.681599] br-lan: port 7(phy0.0-ap0) entered disabled state
Tue Jan  7 09:34:15 2025 kern.info kernel: [   39.687402] mt7996e 0000:01:00.0 phy0.0-ap0: entered allmulticast mode
Tue Jan  7 09:34:15 2025 kern.info kernel: [   39.694104] mt7996e 0000:01:00.0 phy0.0-ap0: entered promiscuous mode
Tue Jan  7 09:34:16 2025 kern.info kernel: [   40.743591] br-lan: port 7(phy0.0-ap0) entered disabled state
Tue Jan  7 09:34:16 2025 kern.info kernel: [   40.749421] mt7996e 0000:01:00.0 phy0.0-ap0: left allmulticast mode
Tue Jan  7 09:34:16 2025 kern.info kernel: [   40.755699] mt7996e 0000:01:00.0 phy0.0-ap0: left promiscuous mode
Tue Jan  7 09:34:16 2025 kern.info kernel: [   40.761921] br-lan: port 7(phy0.0-ap0) entered disabled state
Tue Jan  7 09:34:16 2025 daemon.notice hostapd: nl80211: Failed to remove interface phy0.0-ap0 from bridge br-lan: Invalid argument
Tue Jan  7 09:34:17 2025 daemon.err hostapd: Failed to request a scan of neighboring BSSes ret=-16 (Resource busy) - try to scan again (attempt 3)
Tue Jan  7 09:34:17 2025 kern.info kernel: [   41.835340] br-lan: port 7(phy0.0-ap0) entered blocking state
Tue Jan  7 09:34:17 2025 kern.info kernel: [   41.841090] br-lan: port 7(phy0.0-ap0) entered disabled state
Tue Jan  7 09:34:17 2025 kern.info kernel: [   41.846865] mt7996e 0000:01:00.0 phy0.0-ap0: entered allmulticast mode
Tue Jan  7 09:34:17 2025 kern.info kernel: [   41.853479] mt7996e 0000:01:00.0 phy0.0-ap0: entered promiscuous mode
Tue Jan  7 09:34:17 2025 kern.info kernel: [   42.073343] br-lan: port 1(sfp-lan) entered learning state
(...)
Tue Jan  7 09:34:39 2025 daemon.notice hostapd: nl80211: Failed to remove interface phy0.0-ap0 from bridge br-lan: Invalid argument
Tue Jan  7 09:34:40 2025 daemon.err hostapd: Failed to request a scan of neighboring BSSes ret=-16 (Resource busy) - try to scan again (attempt 14)
Tue Jan  7 09:34:40 2025 kern.info kernel: [   65.225577] br-lan: port 7(phy0.0-ap0) entered blocking state
Tue Jan  7 09:34:40 2025 kern.info kernel: [   65.231331] br-lan: port 7(phy0.0-ap0) entered disabled state
Tue Jan  7 09:34:40 2025 kern.info kernel: [   65.237137] mt7996e 0000:01:00.0 phy0.0-ap0: entered allmulticast mode
Tue Jan  7 09:34:41 2025 kern.info kernel: [   65.243803] mt7996e 0000:01:00.0 phy0.0-ap0: entered promiscuous mode
Tue Jan  7 09:34:42 2025 kern.info kernel: [   66.273729] br-lan: port 7(phy0.0-ap0) entered disabled state
Tue Jan  7 09:34:42 2025 kern.info kernel: [   66.279587] mt7996e 0000:01:00.0 phy0.0-ap0: left allmulticast mode
Tue Jan  7 09:34:42 2025 kern.info kernel: [   66.285868] mt7996e 0000:01:00.0 phy0.0-ap0: left promiscuous mode
Tue Jan  7 09:34:42 2025 kern.info kernel: [   66.292082] br-lan: port 7(phy0.0-ap0) entered disabled state
Tue Jan  7 09:34:42 2025 daemon.notice hostapd: nl80211: Failed to remove interface phy0.0-ap0 from bridge br-lan: Invalid argument
Tue Jan  7 09:34:43 2025 kern.info kernel: [   67.385504] br-lan: port 7(phy0.0-ap0) entered blocking state
Tue Jan  7 09:34:43 2025 kern.info kernel: [   67.391254] br-lan: port 7(phy0.0-ap0) entered disabled state
Tue Jan  7 09:34:43 2025 kern.info kernel: [   67.397035] mt7996e 0000:01:00.0 phy0.0-ap0: entered allmulticast mode
Tue Jan  7 09:34:43 2025 kern.info kernel: [   67.403659] mt7996e 0000:01:00.0 phy0.0-ap0: entered promiscuous mode
Tue Jan  7 09:34:43 2025 kern.info kernel: [   67.532535] br-lan: port 7(phy0.0-ap0) entered blocking state
Tue Jan  7 09:34:43 2025 kern.info kernel: [   67.538287] br-lan: port 7(phy0.0-ap0) entered listening state
```

rany2 commented:
I've reverted the upstream commit causing the issue as it seems like the best solution for now.

AgustinLorenzo commented:
Hi @rany2 

I've added your updates for my next version and we'll see how it goes.

Target: qualcommax/ipq807x

Regards, Agustin :)

rany2 commented:
@AgustinLorenzo your fork seems really popular so that's very helpful in terms of getting more testing, thanks a lot!

e9hack commented:
I got a more useful bt by using hostapd from the build directory instead of the version from staging_dir:
```
hb@z-box:~/src/LEDE/RT-AX53U-6.6.x-test/build_dir/target-mipsel_1004kc_musl/hostapd-full-openssl/hostapd-2025.01.09~d73ad8cb/hostapd$ ~/src/LEDE/RT-AX53U-6.6.x-test/scripts/remote-gdb ~/src/LEDE/RT-AX53U-6.6.x-test/hostapd.1736621008.2502.11.core ./hostapd
Choose target:
  1) mipsel_1004kc (musl )
GNU gdb (GDB) 15.2
Copyright (C) 2024 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "--host=x86_64-pc-linux-gnu --target=mipsel-openwrt-linux-musl".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./hostapd...
[New LWP 2502]
Core was generated by `/usr/sbin/hostapd -s -g /var/run/hostapd/global'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x5560a4a8 in eapol_port_timers_tick (eloop_ctx=0x0, timeout_ctx=0x775e9240) at ../src/eapol_auth/eapol_auth_sm.c:180
180             if (state->eap_if->retransWhile > 0) {
(gdb)
```

danpawlik reviewed (approved):
After few hours -  works normally.

danpawlik commented:
Does someone have an error like:
```
daemon.err hostapd: Failed to create interface phy0.2-ap0: -23 (Too many open files in system)
```

It seems that on BPI-R4, where I have 3 wireless network interfaces, one of them needs to be disabled to avoid such error. 
Adding:
```
cat << 'EOF' >> /etc/sysctl.conf
fs.file-max = 2097152
fs.nr_open = 2097152
EOF
```
does not seems to help.

rany2 commented:
@danpawlik good catch, can you try to add something like `procd_set_param limits nofile="10000 20000"` to the init file?

danpawlik commented:
After adding what you suggest:
```
sed -i 's@procd_set_param limits core="unlimited"@procd_set_param limits core="unlimited"\n                procd_set_param limits nofile="10000 20000"@g' /etc/init.d/wpad
```
still same error. Would be good if someone else with BPi-R4 or 3 wireless interfaces will test

rany2 commented:
Can you create the WiFi interfaces using `iw`? Something like `iw phy <phy name> interface add <interface name> type __ap` to see if you get the same error? Maybe it's a driver issue.

I can reproduce the issue only if the VAP interface name already exists:

```
# iw phy wl0 interface add testap0 type __ap
# iw phy wl0 interface add testap0 type __ap
command failed: Too many open files in system (-23)
# iw dev testap0 del
# iw phy wl0 interface add testap0 type __ap
```

I'd check `ip link` to see if you for some reason you needed to do manually delete the interface for some reason. On my end I don't seem to have this issue.

danpawlik commented:
@rany2 after executing ```iw phy phy0 interface add testap0 type __ap``` nothing happen.
Today after update - all is fine, but I disabled usteer. Probably it was causing an issue. 
Sorry for the noice @rany2 . 

Patch LGTM.

rany2 commented:
@danpawlik I've added you as a Tested-by! Thanks 

Personally I think this PR should be ready. If there are any issues with this update I guess they're very non-obvious.

rany2 commented:
Hey @e9hack ,

Sorry for the ping but I noticed that https://w1.fi/cgit/hostap/commit/?id=d37045e859860a10c37a69f6d7de84c928799d6a seems to address your issue. Can you confirm that this patch actually fixes this issue? If so, then that means I should be able to update to latest master and avoid reverting 01677c47fb13 ("AP: Support disconnect with MLD").

Thanks

rany2 commented:
P.S. you can test the current version, I dropped the revert commit and synced up with hostapd master.

rany2 commented:
It looks like wolfssl is broken with latest hostapd. I'll investigate it later but it might be an upstream issue as they made a lot of wolfssl related changes recently.

danpawlik commented:
@rany2 oh, that's true. I can not create image anymore (I don't know if it is related to wolfssl package update):
```
aarch64-openwrt-linux-musl-gcc -c -o /home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/hostapd-wpad-full-wolfssl/hostapd-2025.02.02~c12fc97e/build/hostapd/src/crypto/crypto_wolfssl.o -I/h
ome/user/openwrt-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/libnl-tiny -I/home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/hostapd-wpad-full-wolfssl/hostapd-20
25.02.02~c12fc97e/src/crypto -I/home/user/openwrt-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_musl/usr/include -I/home/user/openwrt-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_
gcc-13.3.0_musl/include -I/home/user/openwrt-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_musl/include/fortify -DCONFIG_LIBNL20 -D_GNU_SOURCE -DCONFIG_MSG_MIN_PRIORITY=3 -Os -pipe -mcpu=cort
ex-a53 -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/hostapd-wpad-full-wolfssl/hostapd-2025.02.02~c12fc97e=hostapd-2025.02
.02~c12fc97e -ffunction-sections -fdata-sections -flto=auto -fno-fat-lto-objects -Wformat -Werror=format-security -DPIC -fPIC -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/home/user/openwrt-buil
der/openwrt/build_dir/target-aarch64_cortex-a53_musl/hostapd-wpad-full-wolfssl/hostapd-2025.02.02~c12fc97e/src -I/home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/hostapd-wpad-full-wolfss
l/hostapd-2025.02.02~c12fc97e/src/utils -DHOSTAPD -DCONFIG_TAXONOMY -DCONFIG_ELOOP_EPOLL -DUBUS_SUPPORT -DUCODE_SUPPORT -DCONFIG_FULL_DYNAMIC_VLAN -DCONFIG_CTRL_IFACE_MIB -DCONFIG_CTRL_IFACE_UNIX -DCONFIG_CTRL_IFA
CE -DCONFIG_RSN_PREAUTH -DCONFIG_SUITEB192 -DCONFIG_OCV -DCONFIG_IEEE80211R -DCONFIG_IEEE80211R_AP -DCONFIG_ETH_P_OUI -DCONFIG_SAE -DCONFIG_OWE -DCONFIG_AIRTIME_POLICY -DCONFIG_FILS -DCONFIG_WNM -DCONFIG_WNM_AP -D
CONFIG_IEEE80211AC -DCONFIG_IEEE80211BE -DCONFIG_IEEE80211AX -DCONFIG_MBO -DCONFIG_DRIVER_WIRED -DCONFIG_DRIVER_NL80211 -I/home/user/openwrt-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/l
ibnl3 -DCONFIG_DRIVER_WIRED -DCONFIG_DRIVER_NL80211 -I/home/user/openwrt-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/libnl3 -DCONFIG_ERP -DEAP_SERVER_MD5 -DEAP_SERVER_TLS -DEAP_SERVER_PE
AP -DEAP_SERVER_TTLS -DEAP_SERVER_MSCHAPV2 -DEAP_SERVER_GTC -DEAP_SERVER_PSK -DEAP_SERVER_PWD -DEAP_SERVER_FAST -DCONFIG_WPS -DEAP_SERVER_WSC -DCONFIG_WPS_NFC -DCONFIG_WPS_OOB -DEAP_SERVER_IDENTITY -DEAP_SERVER -D
PKCS12_FUNCS -DEAP_TLS_FUNCS -DCONFIG_TLS_WOLFSSL -DCRYPTO_RSA_OAEP_SHA256 -DCONFIG_DES -DCONFIG_SHA256 -DCONFIG_HMAC_SHA256_KDF -DCONFIG_HMAC_SHA384_KDF -DCONFIG_HMAC_SHA512_KDF -DCONFIG_SHA384 -DCONFIG_SHA512 -D
CONFIG_ECC -DCONFIG_NO_RANDOM_POOL -DRADIUS_SERVER -DCONFIG_IPV6 -DNEED_AP_MLME -DCONFIG_HS20 -DCONFIG_INTERWORKING -DCONFIG_PROXYARP -DCONFIG_ACS -DCONFIG_DEBUG_SYSLOG -DCONFIG_APUP -I/home/user/openwrt-builder/o
penwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/libnl-tiny -I/home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/hostapd-wpad-full-wolfssl/hostapd-2025.02.02~c12fc97e/src/cryp
to -I/home/user/openwrt-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_musl/usr/include -I/home/user/openwrt-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_musl/include -I
/home/user/openwrt-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_musl/include/fortify -DCONFIG_LIBNL20 -D_GNU_SOURCE -DCONFIG_MSG_MIN_PRIORITY=3 -I/home/user/openwrt-builder/openwrt/build_dir
/target-aarch64_cortex-a53_musl/hostapd-wpad-full-wolfssl/hostapd-2025.02.02~c12fc97e/src -I/home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/hostapd-wpad-full-wolfssl/hostapd-2025.02.02~
c12fc97e/src/utils -DCONFIG_ELOOP_EPOLL -DUBUS_SUPPORT -DUCODE_SUPPORT -DCONFIG_HT_OVERRIDES -DCONFIG_VHT_OVERRIDES -DCONFIG_BACKEND_FILE -DCONFIG_SUITEB192 -DCONFIG_OCV -DCONFIG_IEEE80211R -DCONFIG_MESH -DCONFIG_
SAE -DCONFIG_OWE -DCONFIG_FILS -DCONFIG_WNM -DCONFIG_IBSS_RSN -DCONFIG_HS20 -DCONFIG_INTERWORKING -DCONFIG_DRIVER_WIRED -DCONFIG_DRIVER_NL80211 -I/home/user/openwrt-builder/openwrt/staging_dir/target-aarch64_corte
x-a53_musl/usr/include/libnl3 -DCONFIG_ERP -DEAP_TLS -DEAP_PEAP -DEAP_TTLS -DEAP_MD5 -DEAP_MSCHAPv2 -DEAP_GTC -DEAP_OTP -DEAP_LEAP -DEAP_PSK -DEAP_FAST -DEAP_PWD -DCONFIG_ECC -DCONFIG_WPS -DEAP_WSC -DCONFIG_WPS_NF
C -DCONFIG_WPS_OOB -DIEEE8021X_EAPOL -DCONFIG_AP -DCONFIG_CTRL_IFACE_MIB -DEAP_SERVER -DEAP_SERVER_IDENTITY -DCONFIG_IEEE80211AC -DCONFIG_IEEE80211BE -DCONFIG_IEEE80211AX -DNEED_AP_MLME -DEAP_SERVER_WSC -DCONFIG_M
BO -DCONFIG_ACS -DPKCS12_FUNCS -DCONFIG_SMARTCARD -DCONFIG_TLS_WOLFSSL -DCRYPTO_RSA_OAEP_SHA256 -DWOLFSSL_DER_LOAD -DCONFIG_DES -DCONFIG_SHA256 -DCONFIG_HMAC_SHA256_KDF -DCONFIG_HMAC_SHA384_KDF -DCONFIG_HMAC_SHA51
2_KDF -DCONFIG_SHA384 -DCONFIG_SHA512 -DCONFIG_ECC -DCONFIG_NO_RANDOM_POOL -DCONFIG_CTRL_IFACE -DCONFIG_CTRL_IFACE_UNIX -DCONFIG_IPV6 -DCONFIG_NO_LINUX_PACKET_SOCKET_WAR -DCONFIG_SME -DCONFIG_DEBUG_SYSLOG -DLOG_HO
STAPD=LOG_DAEMON -DCONFIG_GAS -DCONFIG_OFFCHANNEL ../src/crypto/crypto_wolfssl.c
../src/crypto/crypto_wolfssl.c: In function 'read_rsa_key_from_x509':
../src/crypto/crypto_wolfssl.c:3617:28: error: storage size of 'cert' isn't known
 3617 |         struct DecodedCert cert;
      |                            ^~~~
../src/crypto/crypto_wolfssl.c:3630:46: error: 'NO_VERIFY' undeclared (first use in this function)
 3630 |         err = wc_ParseCert(&cert, CERT_TYPE, NO_VERIFY, NULL);
      |                                              ^~~~~~~~~
../src/crypto/crypto_wolfssl.c:3630:46: note: each undeclared identifier is reported only once for each function it appears in
../src/crypto/crypto_wolfssl.c: In function 'crypto_rsa_key_read':
../src/crypto/crypto_wolfssl.c:3707:42: error: 'RSAk' undeclared (first use in this function)
 3707 |         if (private_key && key_format != RSAk) {
      |                                          ^~~~
make[3]: *** [../src/build.rules:87: /home/user/openwrt-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/hostapd-wpad-full-wolfssl/hostapd-2025.02.02~c12fc97e/build/hostapd/src/crypto/crypto_wolfssl.o] Err
or 1
```

rany2 commented:
I sent a message to the mailing list with a fix. In the meantime I'll update this PR as well.

rany2 commented:
The [build failure for x86/64](https://github.com/openwrt/openwrt/actions/runs/13120684160/job/36605657155?pr=17423) is not related to the changes in this PR (likely caused by commit https://github.com/openwrt/openwrt/commit/73a6cb983c46cbe2e3a603dd5b8ec3155547f8d5).

The [malta build](https://github.com/openwrt/openwrt/actions/runs/13120684160/job/36605657908?pr=17423) appears to have gone well though.

rany2 commented:
> I sent a message to the mailing list with a fix for the wolfssl issue. In the meantime I'll update this PR as well.

This issue is now fixed upstream. Now there are no new patches added by this PR.

AgustinLorenzo commented:
Hi @rany2,

I'm going to compile again, I'll update with your latest changes in my build :)

Regards, Agustin

rany2 committed:
hostapd: update to version 2025-02-09

Manually refreshed:

010-mesh-Allow-DFS-channels-to-be-selected-if-dfs-is-ena.patch
110-mbedtls-TLS-crypto-option-initial-port.patch
140-tests-Makefile-make-run-tests-with-CONFIG_TLS.patch
301-mesh-noscan.patch
601-ucode_support.patch
780-Implement-APuP-Access-Point-Micro-Peering.patch

Dropped upstreamed:

330-nl80211_fix_set_freq.patch
804-hostapd-Fix-clearing-up-settings-for-color-switch.patch

Automatically rebased all other patches.

Tested-by: Rany Hany <rany_hany@riseup.net> # ramips_mt7621/asus_rt-ax53u, mt7622/xiaomi_redmi-router-ax6s
Tested-by: Andre Heider <a.heider@gmail.com> # filogic/openwrt_one, ramips_mt7621/netgear_wac124
Tested-by: Agustin Lorenzo <agustin.lorenzo@thinco.es> # qualcommax/ipq807x (AX3600)
Tested-by: Daniel Pawlik <pawlik.dan@gmail.com> # BPi-R4 with mt7996
Signed-off-by: Rany Hany <rany_hany@riseup.net>

nbd168 commented:
Applied, thanks!
